protocol/assertion.go

```diff
@@ -132,7 +136,16 @@ func (p *ParsedCredentialAssertionData) Verify(storedChallenge string, relyingPa
 
 	sigData := append(p.Raw.AssertionResponse.AuthenticatorData, clientDataHash[:]...)
 
-	key, err := webauthncose.ParsePublicKey(credentialBytes)
+	var (
+		key interface{}
+		err error
+	)
+
+	if appID == "" {
+		key, err = webauthncose.ParsePublicKey(credentialBytes)
+	} else {
+		key, err = webauthncose.ParseFIDOPublicKey(credentialBytes)
+	}
 
```

Comment:
I am using the code to support U2F keys and everything is working. But to be safe the error from `webauthncose.VerifySignature` shouldn't be ignored. There isn't any outcome that causes valid to be `true` with an error but both should be checked with `!valid || err != nil`. The errors from `webauthncose.ParsePublicKey` and `webauthncose.ParseFIDOPublicKey` are also unhandled.

```go
	if appID == "" {
		key, err = webauthncose.ParsePublicKey(credentialBytes)
	} else {
		key, err = webauthncose.ParseFIDOPublicKey(credentialBytes)
	}
	if err != nil {
		return err
	}

	valid, err := webauthncose.VerifySignature(key, sigData, p.Response.Signature)
	if !valid || err != nil {
		return ErrAssertionSignature.WithDetails(fmt.Sprintf("Error validating the assertion signature: %+v\n", err))
	}
```

Also unrelated to this pull request the challenge compare here https://github.com/duo-labs/webauthn/blob/master/protocol/client.go#L80 could be done with constant time compare to prevent any potential timing attacks. Changing `if 0 != strings.Compare(storedChallenge, challenge) {` to `if subtle.ConstantTimeCompare([]byte(storedChallenge), []byte(challenge)) != 1 {`.

Reply:
We can definitely do that, I had in mind to ask about this actually since it seemed odd not to check errors.

Reply:
Good spot on the comparison. This has been added and the other changes have been pushed. If you could check if you think the changes are sufficient (the last two commits) that'd be good.